randovania/layout/description_migration.py

```diff
@@ -601,6 +601,22 @@ def _migrate_v31(data: dict) -> None:
     _migrate_hint_precision(data, {"broad-category"})
 
 
+def _migrate_v32(data: dict) -> None:
+    game_modifications = data["game_modifications"]
+
+    for game in game_modifications:
+        game_name = game["game"]
+        if game_name != "samus_returns":
+            continue
+        locations = game.get("locations")
+
+        old_new_name = migration_data.get_raw_data(RandovaniaGame(game_name))["a4_crystal_mines_typo"]
+        for old_name, new_name in old_new_name.items():
+            for area, nodes in locations.items():
+                if old_name in area:
```

Comment:
I'm fairly certain the reason of your failing tests is this line.
your migration data only contains nodes.
But you need to rename:
- all places where nodes in the room are mentioned
- all places where the room name itself is mentioned.

At first glance that comes to mind for that is:
- item locations
- DLR - but the room doesnt have any doors
- generation order
- starting room
- excluded locations
Maybe something more? Maybe some of those are not valid.